packages/twenty-server/src/engine/core-modules/email/email-sender.service.ts

```diff
@@ -1,16 +1,88 @@
-import { Inject, Injectable } from '@nestjs/common';
+import { Injectable } from '@nestjs/common';
 
 import { SendMailOptions } from 'nodemailer';
 
-import { EmailDriver } from 'src/engine/core-modules/email/drivers/interfaces/email-driver.interface';
+import { EmailDriverInterface } from 'src/engine/core-modules/email/drivers/interfaces/email-driver.interface';
 
-import { EMAIL_DRIVER } from 'src/engine/core-modules/email/email.constants';
+import { LoggerDriver } from 'src/engine/core-modules/email/drivers/logger.driver';
+import { SmtpDriver } from 'src/engine/core-modules/email/drivers/smtp.driver';
+import { EmailDriver } from 'src/engine/core-modules/email/enums/email-driver.enum';
+import { DynamicDriverBase } from 'src/engine/core-modules/twenty-config/dynamic-driver.base';
+import { ConfigVariablesGroup } from 'src/engine/core-modules/twenty-config/enums/config-variables-group.enum';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
 
 @Injectable()
-export class EmailSenderService implements EmailDriver {
-  constructor(@Inject(EMAIL_DRIVER) private driver: EmailDriver) {}
+export class EmailSenderService
```

Comment:
I'm a bit lost between EmailDriverInterface, DynamicDriverBase and EmailSenderService namings. Based on what I see in the file a more classic naming would be a Factory (a class returning a class depending on configuration)

Also, not sure we need this DynamicD

Reply:
my comments are non blocking, I think this stays maintainable and it's better than what we have (non dynamic). Just adding my two cents